src/runtime_src/core/edge/drm/zocl/zocl_xclbin.c

```diff
@@ -992,29 +1013,27 @@ zocl_load_aie_only_pdi(struct drm_zocl_dev *zdev, struct axlf *axlf,
  * @param       slot: Specific slot structure
  *
  */
-void zocl_free_sections(struct drm_zocl_slot *slot)
+void zocl_free_sections(struct drm_zocl_dev *zdev, struct drm_zocl_slot *slot)
 {
-	if (slot->ip) {
+	if (slot->ip)
 		vfree(slot->ip);
-		CLEAR(slot->ip);
-	}
-	if (slot->debug_ip) {
+	if (slot->debug_ip)
 		vfree(slot->debug_ip);
-		CLEAR(slot->debug_ip);
-	}
-	if (slot->connectivity) {
+	if (slot->connectivity)
 		vfree(slot->connectivity);
-		CLEAR(slot->connectivity);
-	}
-	if (slot->topology) {
+	if (slot->topology)
 		vfree(slot->topology);
-		CLEAR(slot->topology);
-	}
-	if (slot->axlf) {
+	if (slot->axlf)
 		vfree(slot->axlf);
```

Comment:
vfree can ignore NULL pointer. Maybe we don't need to check if it is NULL?

See mm/vmalloc.c:
/**
 * vfree - Release memory allocated by vmalloc()
 * @addr:  Memory base address
 *
 * Free the virtually continuous memory area starting at @addr, as obtained
 * from one of the vmalloc() family of APIs.  This will usually also free the
 * physical memory underlying the virtual allocation, but that memory is
 * reference counted, so it will not be freed until the last user goes away.
 *
 * If @addr is NULL, no operation is performed.
 *
 * Context:
 * May sleep if called *not* from interrupt context.
 * Must not be called in NMI context (strictly speaking, it could be
 * if we have CONFIG_ARCH_HAVE_NMI_SAFE_CMPXCHG, but making the calling
 * conventions for vfree() arch-dependent would be a really bad idea).
 */
 void vfree(const void *addr)